packages/doenetml-worker-rust/lib-doenetml-core/src/utils/keyvalue.rs

```diff
@@ -19,4 +20,18 @@ where
 
         None
     }
+
+    /// Match key to the HashMap keys, ignoring case.
+    /// If found, remove the original value from the HashMap.
+    fn remove_ignore_case(&mut self, key: &str) -> Option<V> {
```

Comment:
I don't know how this is used, but it smells of quadratic behavior somewhere...You might consider creating a hashmap of normalized names and a normalization function.

Reply:
This is used for one case only: We start with the original attributes, called `unevaluated_attributes` which is the hashmap that we get from the dast. Then, if the component type specifies that a particular attribute, it is removed from `unevaluated_attributes` with a case-insensitive match (function `evaluate_attributes_replace_macros`  of `component_creation.rs`).

Rust borrowing rules kept all my first algorithms from working, so I ended up with this one. More than happy to replace it with a better algorithm if this one is suspect. I don't care if it is a general purpose function on a HashMap like this or just programmed specifically in `evaluate_attributes_replace_macros`.